avm/ptn/ai-platform/baseline/main.bicep

```diff
@@ -0,0 +1,331 @@
+
+@allowed([
+  'disabled'
+  'enabled'
+])
+@description('Optional. Whether the trust policy is enabled for the container registry.')
+param containerRegistryTrustPolicyStatus string = 'enabled'
+
+@description('Optional. Provide \'true\' to enable Key Vault\'s purge protection feature.')
+param keyVaultEnablePurgeProtection bool = true
+
+@description('Optional. Indicates whether the storage account permits requests to be authorized with the account access key via Shared Key. If false, then all requests, including shared access signatures, must be authorized with Azure Active Directory (Azure AD). The default value is null, which is equivalent to true.')
+param storageAccountAllowSharedKeyAccess bool = false
+
+@sys.description('Optional. Computes to create and attach to the workspace hub.')
+param workspaceComputes array = []
+
+@allowed([
+  'AllowInternetOutbound'
+  'AllowOnlyApprovedOutbound'
+])
+@description('Optional. The network isolation mode of the workspace hub.')
+param workspaceNetworkIsolationMode string = 'AllowInternetOutbound'
+
+@description('Optional. The outbound rules for the managed network of the workspace hub.')
+param workspaceNetworkOutboundRules workspaceNetworkOutboundRuleType
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.aiplatform-baseline.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: 'id-${name}'
+  location: location
+  tags: tags
+}
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2023-09-01' = {
+  name: 'log-${name}'
+  location: location
+  tags: tags
+}
+
+module keyVault 'br/public:avm/res/key-vault/vault:0.6.2' = {
+  name: '${uniqueString(deployment().name, location)}-key-vault'
+  params: {
+    name: 'kv-${name}'
+    location: location
+    enableTelemetry: enableTelemetry
+    enableRbacAuthorization: true
+    enableVaultForDeployment: false
+    enableVaultForDiskEncryption: false
+    enableVaultForTemplateDeployment: false
+    networkAcls: {
+      defaultAction: 'Deny'
+      bypass: 'AzureServices'
+    }
+    publicNetworkAccess: 'Disabled'
+    enablePurgeProtection: keyVaultEnablePurgeProtection
+    roleAssignments: [
+      {
+        principalId: managedIdentity.properties.principalId
+        roleDefinitionIdOrName: 'Key Vault Administrator'
```

Comment:
Is there something smaller like `Key Vault Secrets Officer` ?

Reply:
https://github.com/Azure/bicep-registry-modules/pull/2694#discussion_r1672444855

Reply:
I think we should leave this one